development/frontend/testing.md

```diff
@@ -0,0 +1,73 @@
+## Front-End Testing
+
+For our front-end we test on two fronts:
+
+- [Unit Tests](#unit-tests) - Focusses on individual functional testing. Each function should be tested in complete isolation
+- [E2E Integration Tests](#e2e-integration-tests) - Driven by automated testing framework, [Cypress](https://www.cypress.io/), this tests our front-end in it's entirety, including button clicks, navigations and API calls.
+
+## Unit Tests
+
+Not yet written
+
+## E2E Integration Tests
+
+### Running Locally
+
+To run the E2E tests yourself, we need to first run our own local web server that the tests will run on. For our testing, we have a single server configured with multiple users, teams and projects for testing purposes. To run this, open a terminal and run:
+
+```bash
+npm run cy:web-server
+```
+
+Once this is up and running, you then have two options:
+
+#### Run Tests via Terminal
+
+```bash
+npm run cy:run
+```
+
+Simply run, in a terminal, all of the front-end E2E tests and displays the results. If you just want to run all the tests, and aren't necessarily developing your own, this is the command to run.
```

Comment:
```suggestion

This will execute all of the front-end E2E tests in the terminal and display the results. 
Typically, this is the command to run if you simply want to run all of the tests
When developing/debugging tests, opening the Cypress Application will likely be more useful.
```